src/SDK/ConfigBuilder.php

```diff
@@ -0,0 +1,199 @@
+        'OTEL_EXPORTER_OTLP_TIMEOUT' => 'trace.exporters.otlp.timeout',
+        'OTEL_EXPORTER_OTLP_TRACES_TIMEOUT' => 'trace.exporters.otlp.timeout',
+        'OTEL_EXPORTER_OTLP_PROTOCOL' => 'trace.exporters.otlp.protocol',
+        'OTEL_EXPORTER_OTLP_TRACES_PROTOCOL' => 'trace.exporters.otlp.protocol',
+        'OTEL_EXPORTER_JAEGER_AGENT_HOST' => 'trace.exporters.jaeger.agent_host',
+        'OTEL_EXPORTER_JAEGER_AGENT_PORT' => 'trace.exporters.jaeger.agent_port',
+        'OTEL_EXPORTER_JAEGER_ENDPOINT' => 'trace.exporters.jaeger.endpoint',
+        'OTEL_EXPORTER_JAEGER_TIMEOUT' => 'trace.exporters.jaeger.timeout',
+        'OTEL_EXPORTER_JAEGER_USER' => 'trace.exporters.jaeger.user',
+        'OTEL_EXPORTER_JAEGER_PASSWORD' => 'trace.exporters.jaeger.password',
+        'OTEL_PROPAGATORS' => 'propagators',
+        'OTEL_TRACES_EXPORTER' => 'trace.exporter',
+        'OTEL_METRICS_EXPORTER' => 'metrics.exporter',
+        'OTEL_PHP_TRACES_PROCESSOR' => 'trace.processor',
+    ];
+
+    public function build(array $data = []): object
+    {
+        //@phan-ignore PhanUndeclaredClassMethod
+        $schema = Expect::structure([
+            'log' => Expect::structure([
+                'stream' => Expect::string('php://stdout'),
+                'level' => Expect::string('info'),
+            ]),
+            'propagators' => Expect::string('tracecontext,baggage'),
+            'service' => Expect::structure([
+                'name' => Expect::string(),
+            ]),
+            'resource' => Expect::structure([
+                'limits' => Expect::structure([
+                    'attribute_count' => Expect::string()->castTo('int'),
+                    'attribute_value_length' => Expect::string()->castTo('int'),
+                ]),
+                'attributes' => Expect::array(),
+            ]),
+            'trace' => Expect::structure([
+                'sampler' => Expect::string('parentbased_always_on'),
+                'samplers' => Expect::structure([
+                    'traceidratio' => Expect::structure([
+                        'probability' => Expect::string()->castTo('float'),
+                    ]),
+                ]),
+                'processor' => Expect::string('batch'),
+                'processors' => Expect::structure([
+                    'batch' => Expect::structure([
+                        'schedule_delay' => Expect::string()->castTo('int'),
+                        'export_timeout' => Expect::string()->castTo('int'),
+                        'max_queue_size' => Expect::string()->castTo('int'),
+                        'max_export_batch_size' => Expect::string()->castTo('int'),
+                    ]),
+                    'simple' => Expect::structure([]),
+                    'noop' => Expect::structure([]),
+                ]),
+                'exporter' => Expect::string('grpc'),
+                'exporters' => Expect::structure([
+                    'zipkin' => Expect::structure([
```

Comment:
I would prefer to not have the schemas for the contrib exporters here. This is of course not a direct dependency on the Contrib "package", but a indirect one. It's expecting something which may not be there.

Also this looks like it's always expecting configuration for all possible exporters, no?

Reply:
Yes, that makes sense. It does raise the question though - I'm starting to think that the two otlp exporters should not be in `contrib` but rather in `SDK` - this is the opentelementry project, after all.

Reply:
>  this is the opentelementry project, after all.

That's true, however from the perspective of the library the OTLP collectors are 3rd party applications, which just happen to live under the same umbrella organization. Moving the grpc exporter into the SDK would also make the corresponding extension a must have dpendency for the SDK again. And since there must be a solution to handle Zipkin & Jaeger and Co. in the contrin directory, it's not much of a hassle to do the same for the otlp exporters.